### PR TITLE
[dashboard] Fix parsing of contextUrl to handle env vars well

### PR DESCRIPTION
**Pull request by geropl**
Fixes https://github.com/gitpod-io/gitpod/issues/3082.

## Test
 - open https://gpl-3082-gitpod-host-url.staging.gitpod-dev.com/#passedin=test%20value/https://github.com/gitpod-io/gitpod-test-repo
 - wait until workspace started up, then click your account icon (top right) and choose "Stop Workspace"
 - note how you don't stuck on a blank page

### CODE REVIEW

File: components/theia/packages/gitpod-extension/src/browser/githoster/git-state.ts [hunk 1]
```diff
@@ -199,7 +200,7 @@ export abstract class GitState {
             }
         }
         if (CommitContext.is(ws.workspace.context)) {
-            const url = new URL(ws.workspace.contextURL);
+            const url = contextUrlToUrl(ws.workspace.contextURL);
```
Comment (AlexTugarev) [inline components/theia/packages/gitpod-extension/src/browser/githoster/git-state.ts:203]: 👍🏻

File: components/gitpod-protocol/src/util/context-url.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+/**
+ * The field "contextUrl" might contain prefixes like:
+ *  - envvar1=value1/...
+ *  - prebuild/...
+ * This is the analogon to the (Prefix)ContextParser structure in "server".
+ */
+export function contextUrlToUrl(contextUrl: string): URL {
```
Comment (AlexTugarev) [inline components/gitpod-protocol/src/util/context-url.ts:13]: given the call sites it should be:
`contextUrlToUrl(contextUrl: string | undefined): URL | undefined`

**Review by AlexTugarev — APPROVED** [review]
It works as advertised 🙏🏻

File: components/gitpod-protocol/src/util/context-url.ts (hunk 2 above)
Comment (geropl) [reply components/gitpod-protocol/src/util/context-url.ts]: Absolutely correct. Thank you!

**Pull request merged**